src/core/auth/qgsauthmanager.cpp

```diff
@@ -0,0 +1,3291 @@
+    dumpIgnoredSslErrorsCache_();
+    return true;
+  }
+
+  QgsDebugMsg( "Rebuild of ignored SSL errors cache SAME AS BEFORE" );
+  dumpIgnoredSslErrorsCache_();
+  return true;
+}
+
+
+bool QgsAuthManager::storeCertAuthorities( const QList<QSslCertificate> &certs )
+{
+  if ( certs.size() < 1 )
+  {
+    QgsDebugMsg( "Passed certificate list has no certs" );
+    return false;
+  }
+
+  Q_FOREACH ( const QSslCertificate& cert, certs )
+  {
+    if ( !storeCertAuthority( cert ) )
+      return false;
+  }
+  return true;
+}
+
+bool QgsAuthManager::storeCertAuthority( const QSslCertificate& cert )
+{
+  // don't refuse !cert.isValid() (actually just expired) CAs,
+  // as user may want to ignore that SSL connection error
+  if ( cert.isNull() )
+  {
+    QgsDebugMsg( "Passed certificate is null" );
+    return false;
+  }
+
+  removeCertAuthority( cert );
+
+  QString id( QgsAuthCertUtils::shaHexForCert( cert ) );
+  QString pem( cert.toPem() );
+
+  QSqlQuery query( authDbConnection() );
+  query.prepare( QString( "INSERT INTO %1 (id, cert) "
+                          "VALUES (:id, :cert)" ).arg( authDbAuthoritiesTable() ) );
+
+  query.bindValue( ":id", id );
+  query.bindValue( ":cert", pem );
+
+  if ( !authDbStartTransaction() )
+    return false;
+
+  if ( !authDbQuery( &query ) )
+    return false;
+
+  if ( !authDbCommit() )
+    return false;
+
+  QgsDebugMsg( QString( "Store certificate authority SUCCESS for id: %1" ).arg( id ) );
+  return true;
+}
+
+const QSslCertificate QgsAuthManager::getCertAuthority( const QString &id )
+{
+  QSslCertificate emptycert;
+  QSslCertificate cert;
+  if ( id.isEmpty() )
+    return emptycert;
+
+  QSqlQuery query( authDbConnection() );
+  query.prepare( QString( "SELECT cert FROM %1 "
+                          "WHERE id = :id" ).arg( authDbAuthoritiesTable() ) );
+
+  query.bindValue( ":id", id );
+
+  if ( !authDbQuery( &query ) )
+    return emptycert;
+
+  if ( query.isActive() && query.isSelect() )
+  {
+    if ( query.first() )
+    {
+      cert = QSslCertificate( query.value( 0 ).toByteArray(), QSsl::Pem );
+      QgsDebugMsg( QString( "Certificate authority retrieved for id: %1" ).arg( id ) );
+    }
+    if ( query.next() )
+    {
+      QgsDebugMsg( QString( "Select contains more than one certificate authority for id: %1" ).arg( id ) );
+      emit messageOut( tr( "Authentication database contains duplicate certificate authorities" ), authManTag(), WARNING );
+      return emptycert;
+    }
+  }
+  return cert;
+}
+
+bool QgsAuthManager::existsCertAuthority( const QSslCertificate& cert )
+{
+  if ( cert.isNull() )
+  {
+    QgsDebugMsg( "Passed certificate is null" );
+    return false;
+  }
+
+  QString id( QgsAuthCertUtils::shaHexForCert( cert ) );
+
+  QSqlQuery query( authDbConnection() );
+  query.prepare( QString( "SELECT value FROM %1 "
+                          "WHERE id = :id" ).arg( authDbAuthoritiesTable() ) );
+
+  query.bindValue( ":id", id );
+
+  
```

Comment:
Hmm. Not seeing the issue here. Both the parent function and `QgsAuthCertUtils::shaHexForCert` reference `const QSslCertificate &` values.